toolkit/scripts/toolchain/download_toolchain_rpm.sh

```diff
@@ -178,25 +184,45 @@ function download() {
     return 1
 }
 
+# Validate the signatures of the downloaded RPM against the provided GPG keys
 function validate_signatures() {
     echo "Validating toolchain RPM: $rpm_name" | tee -a "$log_file"
 
-    for key in $allowable_gpg_keys; do
-        echo "Adding key ($key) to empty workdir at ($work_dir)" >> "$log_file"
-        rpmkeys --root "$work_dir" --import "$key" >> "$log_file"
-    done
-
     if ! rpmkeys --root "$work_dir" --checksig --verbose "$dst_file" -D "%_pkgverify_level signature" >> "$log_file"; then
         echo "Failed to validate toolchain package $rpm_name signature, aborting." | tee -a "$log_file"
         return 1
     fi
     return 0
 }
 
-mkdir -p "$(dirname "$log_file")"
+# Validate the integrity of the downloaded RPM using its digests
+function validate_hash() {
+    echo "Checking hash of toolchain RPM: $rpm_name" | tee -a "$log_file"
+
+    if ! rpm --root "$work_dir" --checksig --verbose "$dst_file" -D "%_pkgverify_level digest" >> "$log_file"; then
+        echo "Failed to validate toolchain package $rpm_name hashes, aborting." | tee -a "$log_file"
+        return 1
+    fi
+    return 0
```

Comment:
Might be nice to add a statement here that the signature validated ok.  I realize no error is a good result, but if we ever have to research it, it's an easier confirmation when looking at the log file.

Reply:
Sure thing; will add

Reply:
On second thought, I want to avoid extra messages from this script to avoid cluttering up the logs. This script is called for each toolchain package (~267 of them)

Reply:
The output currently looks like this:
```
Checking hash of toolchain RPM: tar-debuginfo-1.35-1.azl3.x86_64.rpm
Downloading toolchain RPM: texinfo-debuginfo-7.0.3-1.azl3.x86_64.rpm
Downloading toolchain RPM: unzip-6.0-21.azl3.x86_64.rpm
Checking hash of toolchain RPM: tdnf-devel-3.5.8-7.azl3.x86_64.rpm
Downloading toolchain RPM: unzip-debuginfo-6.0-21.azl3.x86_64.rpm
Checking hash of toolchain RPM: tdnf-plugin-repogpgcheck-3.5.8-7.azl3.x86_64.rpm
Checking hash of toolchain RPM: tdnf-plugin-metalink-3.5.8-7.azl3.x86_64.rpm
Checking hash of toolchain RPM: tdnf-python-3.5.8-7.azl3.x86_64.rpm
Downloading toolchain RPM: util-linux-2.40.2-1.azl3.x86_64.rpm
```